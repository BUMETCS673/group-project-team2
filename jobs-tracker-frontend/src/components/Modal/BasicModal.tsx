import React from 'react'
import Typography from '@mui/material/Typography';
import Modal from '@mui/material/Modal';
import {StyledBox,  StyledModalHeader, StyledModalFooter, StyledAddBtn, StyledCloseButton, StyledSaveBtn} from '../../styles/styles'
import { MdAdd } from 'react-icons/md'
import { IoCloseCircleOutline } from "react-icons/io5";

const BasicModal = () => {
    const [open, setOpen] = React.useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);
    return (
        <div>
            {/*<AddBtn onClick = {handleOpen}/>*/}
            <StyledAddBtn onClick = {handleOpen}>
                <p>New</p>
                <MdAdd size={25} />
            </StyledAddBtn>
            <Modal
                open = {open}
                onClose = {handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <StyledBox>
                    <StyledModalHeader >
                        <Typography id="modal-modal-title" variant="h6" component="h2">
                            Text in a modal
                        </Typography>
                        <StyledCloseButton onClick = {handleClose}>
                            <IoCloseCircleOutline size = {30} color= 'white'/>
                        </StyledCloseButton>
                    </StyledModalHeader>
                    
                    <Typography id="modal-modal-description" sx={{ m: 2 }}>
                        Duis mollis, est non commodo luctus, nisi erat porttitor ligula.
                    </Typography>
                    <StyledModalFooter>
                        <StyledSaveBtn>
                            <p>
                                Save
                            </p>
                        </StyledSaveBtn>
                        
                    </StyledModalFooter>

                </StyledBox>


            </Modal>
        </div>

    )
}
export default BasicModal