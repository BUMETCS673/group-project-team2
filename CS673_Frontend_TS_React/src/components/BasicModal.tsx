import React from 'react'
import Typography from '@mui/material/Typography'
import {
  StyledBox,
  StyledModalHeader,
  StyledCloseButton,
} from '../styles/styles'
import { IoCloseCircleOutline } from 'react-icons/io5'
import Modal from '@mui/material/Modal'
import { Button } from '@mui/material'
import AddIcon from '@mui/icons-material/Add'
// import { useAppDispatch, useAppSelector } from '../app/hooks'
// import { openBasicModal, closeBasicModal } from '../features/UI/ui_slice'
//import { useState } from 'react'
 import JobForm from './Job_Components/JobForm'
 import ActivityForm from './Activity_Components/ActivityForm'
 import EditIcon from '@mui/icons-material/Edit';
 import useMediaQuery from '@mui/material/useMediaQuery';
 import {Activity} from '../types/types'

type ModalProps = {
  type: string 
  title: string
  buttonTitle: string
  job_id?: string | undefined
  insideCard?: boolean
  priority? : number
  currentActivity? : Activity,
 
}

const BasicModal: React.FC<ModalProps> = ({
  type,
  title,
  buttonTitle,
  job_id,
  insideCard,
  priority,
  currentActivity,
  
}: ModalProps) => {
  const [open, setOpen] = React.useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);
    const matches = useMediaQuery("(min-width:600px)");

  return (
    <div>
      <Button 
        variant="contained" 
        onClick={handleOpen} 
        data-testid = "modal-btn" 
        
        sx = {{
          backgroundColor: "rgb(23, 160, 160)",
          ...(insideCard && priority == 1 && {
            color: 'rgb(77, 77, 77)',
            backgroundColor: "rgb(230, 99, 99)",
          }),
          ...(insideCard && priority == 2 && {
            color: 'rgb(77, 77, 77)',
            backgroundColor: "rgb(216, 186, 88)",
          }),
          ...(insideCard && priority == 3 && {
            color: 'rgb(77, 77, 77)',
            backgroundColor: "rgb(100, 190, 115)",
          }),
          
        }}
        >
        {title == "add" ? (<> 
          <AddIcon />
          <Typography>
            {buttonTitle}
          </Typography>
        </>) : ( <>
          <EditIcon/> 
          {matches && (
            <Typography>
            {buttonTitle}
          </Typography>
          )}
        </>) }
      </Button>

      <Modal
        open={open}
        onClose={handleOpen}
        title="modal"
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <StyledBox>
          <StyledModalHeader>
            <Typography id="modal-modal-title" variant="h6" component="h2">
              {title}
            </Typography>
            <StyledCloseButton onClick={handleClose} title="close-modal" data-testid = "close-modal-btn">
              <IoCloseCircleOutline size={30} color="white" />
            </StyledCloseButton>
          </StyledModalHeader>

          {/* <JobForm /> */}
          {type == 'job' ? <JobForm job_id = {job_id} closePopup = {handleClose} /> : <ActivityForm job_id = {job_id} closePopup = {handleClose} currentActivity = {currentActivity}/> }
        </StyledBox>
      </Modal>
    </div>
  )
}
export default BasicModal
