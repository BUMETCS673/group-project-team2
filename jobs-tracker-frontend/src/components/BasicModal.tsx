import React from 'react'
import Typography from '@mui/material/Typography';
import {StyledBox,  StyledModalHeader, StyledAddBtn, StyledCloseButton} from '../styles/styles'
import { MdAdd } from 'react-icons/md'
import { IoCloseCircleOutline } from "react-icons/io5";
import Modal from '@mui/material/Modal';
import JobForm from "./JobForm"


const BasicModal = () => {
    const [open, setOpen] = React.useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);
    return (
        <div>
            <StyledAddBtn  onClick = {handleOpen} title = "add">
                <p>New</p>
                <MdAdd size={25} />
            </StyledAddBtn>
            <Modal
                open = {open}
                onClose = {handleClose}
                title = "modal"
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <StyledBox>
                    <StyledModalHeader>
                        <Typography id="modal-modal-title" variant="h6" component="h2">
                            Text in a modal
                        </Typography>
                        <StyledCloseButton onClick = {handleClose} title = "close-modal">
                            <IoCloseCircleOutline size = {30} color= 'white' />
                        </StyledCloseButton>
                    </StyledModalHeader>
                    
                    <JobForm/>

                        
                </StyledBox>
            </Modal>    
        </div>

    )
}
export default BasicModal