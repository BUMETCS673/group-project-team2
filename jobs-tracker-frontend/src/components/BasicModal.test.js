import React from 'react';
import { render, screen, fireEvent, waitFor} from '@testing-library/react'
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'
import BasicModal from './BasicModal';


describe("BasicModal component", () => {
    it('renders without crashing', () => {
        const component = renderer.create(
            <BasicModal/>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    });
    
    it('has an add button that opens a modal',  () => {
        render(<BasicModal/>)
        const button = screen.getByRole('button', {name: /New/i})
        fireEvent.click(button)
        const saveBtn = screen.getByRole('button', {name: /Save/i})
        expect(saveBtn).toBeInTheDocument()

        const closeBtn = screen.getByRole('button', {name: /close/i})
        fireEvent.click(closeBtn)
        expect(saveBtn).not.toBeInTheDocument()
    })
    it('renders and submits a formik form', async => {
        const handleSubmit = jest.fn()
        render(<BasicModal onSubmit={handleSubmit} />)
        const user = fireEvent.setup()
      
        await user.type(screen.getByLabelText(/companyName/i), 'Amazon')
        await user.type(screen.getByLabelText(/jobTitle/i), 'Software Engineer')
      
        await user.click(screen.getByRole('button', {name: /save/i}))
      
        await waitFor(() =>
          expect(handleSubmit).toHaveBeenCalledWith({
            companyName: 'Amazon',
            jobTitle: 'Software Engineer',
          }),
        )
      })
    
})