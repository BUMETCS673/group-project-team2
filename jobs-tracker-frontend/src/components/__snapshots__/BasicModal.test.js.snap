// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicModal component renders without crashing 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    <p>
      New
    </p>
    <svg
      fill="currentColor"
      height={25}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={25}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </button>
</div>
`;
